src/mui-themeable.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import DefaultRawTheme from './styles/raw-themes/light-raw-theme';
+import ThemeManager from './styles/theme-manager';
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+export default function muiThemeable(WrappedComponent) {
+  const MuiComponent = (props, {muiTheme = ThemeManager.getMuiTheme(DefaultRawTheme)}) => {
+    return <WrappedComponent {...props} muiTheme={muiTheme} />;
+  };
+
+  MuiComponent.displayName = `mui(${getDisplayName(WrappedComponent)})`;
```

Comment:
I'm going to propose this again. Isn't it better to inverse the control? I mean instead of having parents looking for particular children, they provide a contextual interface for the children to subscribe themselves. This will allow arbitrary nesting of related components without breaking logic. I'll provide a Proposal PR, (like the ones @newoga does) to clarify my point. It will be with the DropDownMenu.